props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+  }
+
+  void TurnBladeOn(BladeSet target_blade) {
+    // Add in all non-controlled blades, effectively excluding the "other" blade.
+    target_blade = target_blade | ~controlled_blades_;
+    if (SaberBase::OnBlades().off()) {
+      PVLOG_NORMAL << "** No blades are currently ON, turning on the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND")
+                   << " blade and all others, excluding the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "SECOND" : "MAIN")
+                   << " blade\n";
+      if (isPointingUp() || muted_) {
+        FastOn(EffectLocation(0, target_blade));
+     } else {
+        On(EffectLocation(0, target_blade));
+      }
+    } else {
+      PVLOG_NORMAL << "** Turning on the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND")
+                   << " blade\n";
+      SaberBase::TurnOn(EffectLocation(0, target_blade));
+    }
+  }
+
+  void TurnBladeOff(BladeSet target_blade) {
+    SaberBase::OffType off_type = isPointingUp() ? SaberBase::OFF_FAST : SaberBase::OFF_NORMAL;
+    PVLOG_DEBUG << "***** Off type: " << (off_type == SaberBase::OFF_FAST ? "OFF_FAST" : "OFF_NORMAL") << "\n";
+
+    // Check if this is the only blade ON (of MAIN or SECOND blades)
+    if ((SaberBase::OnBlades() & ~target_blade & controlled_blades_).off()) {
+      // Other blade is not on, so just do normal Off() with appropriate off_type
+      PVLOG_NORMAL << "** Turning OFF all blades\n";
+      Off(off_type);
+      muted_ = false;
+    } else {
+      // Only Turn OFF this blade, leave the other one ON.
+      PVLOG_NORMAL << "** Turning OFF only the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND") 
+                   << " blade\n";
+      SaberBase::TurnOff(off_type, EffectLocation(0, target_blade));
+    }
+    battle_mode_ = false;
+    spam_blast_ = false;
+    saber_off_time_ = millis();
+  }
+
+  void TurnMainBladeOnMuted() {
+    if (hybrid_font.PlayPolyphonic(&SFX_mute)) {
+      mute_mainBlade_timer_.trigger(SaberBase::sound_length * 1000);
+    } else {
+      mute_mainBlade_timer_.trigger(0);
+    }
+  }
+
+  void TurnSecondBladeOnMuted() {
+    if (hybrid_font.PlayPolyphonic(&SFX_mute)) {
+      mute_secondBlade_timer_.trigger(SaberBase::sound_length * 1000);
+    } else {
+      mute_mainBlade_timer_.trigger(0);
+    }
+  }
+
+  void TurnDualBladesOn() {
+    TurnOnHelper();
+    PVLOG_NORMAL << "**** Dual Blades Turned ON\n";
+  }
+
+  void TurnDualBladesOff() {
+    TurnOffHelper();
+    PVLOG_NORMAL << "** Dual Blades Turned OFF\n";
+  }
+
+  // Determine the active blade based on x-axis motion - for thrust effects
+  void GetThrustBladeLocation() {
+    mss = fusor.mss();
+    if (mss.x < 14) {
+        thrusting_blade_ = BC_MAIN_BLADE_SET;
+    } else if (mss.x > -14) {
+        thrusting_blade_ = BC_SECOND_BLADE_SET;
+    }
+    thrust_location = EffectLocation(0, thrusting_blade_);
```

Comment:
Why store this in a variable, why not return it?

Reply:
ah yes. 
ok done